scripts/benchmark_all.sh

```diff
@@ -0,0 +1,153 @@
+
+    echo "Running $benchmark for $commit... (multi-threaded)"
+    ( ${build_folder}/$benchmark --scheduler --clients ${num_mt_clients} -o "${build_folder}/benchmark_all_results/${benchmark}_${commit}_mt.json" 2>&1 ) | tee "${build_folder}/benchmark_all_results/${benchmark}_${commit}_mt.log"
+  done
+  cd "${build_folder}"
+
+  # After all benchmarks are done, leave a marker so that this commit can be skipped next time
+  touch "benchmark_all_results/complete_${commit}"
+done
+
+# Print the results
+
+echo ""
+echo "==========="
+echo ""
+
+# Print information about the system
+echo "**System**"
+echo "<details>"
+echo "<summary>$(hostname) - click to expand</summary>"
+echo ""
+echo "| property | value |"
+echo "| -- | -- |"
+echo "| Hostname | $(hostname) |"
+if [[ "$(uname)" == 'Darwin' ]]; then
+  # Retrieve the model name. Might be more than one line if the hardware has been updated.
+  echo "| Model | $(defaults read ~/Library/Preferences/com.apple.SystemProfiler.plist 'CPU Names' | cut -sd '"' -f 4 | tail -n 1) |"
+  echo "| CPU | $(sysctl -n machdep.cpu.brand_string) |"
+  echo "| Memory | $(system_profiler SPHardwareDataType | grep '  Memory:' | sed 's/Memory://') |"
+else
+  echo "| CPU | $(lscpu | grep 'Model name' | sed 's/Model name: *//') |"
+  echo "| Memory | $(free --si -h | grep Mem | awk '{ print $4 "B" }') |"
+  if [ -x /usr/bin/numactl ]; then
+    for bind_type in nodebind membind
+    do
+      echo "| numactl | $(numactl --show | grep --color=never ${bind_type}) |"
+    done
+  else
+    echo "| numactl | binary not found |"
+  fi
+fi
+echo "</details>"
+
+# Print information about the time spent building the commits
+echo ""
+echo "**Build Time**"
+echo "| commit | build time |"
+echo "| -- | -- |"
+echo -n "| ${start_commit} |"
+cat "${build_folder}/benchmark_all_results/build_time_${start_commit}.txt" | xargs | awk '{printf $0 "|\n"}'
+echo -n "| ${end_commit} |"
+cat "${build_folder}/benchmark_all_results/build_time_${end_commit}.txt" | xargs | awk '{printf $0 "|\n"}'
+
+# Print information for each benchmark
+for benchmark in $benchmarks
+do
+  for config in st mt
+  do
+    output=$(../scripts/compare_benchmarks.py "${build_folder}/benchmark_all_results/${benchmark}_${start_commit}_${config}.json" "${build_folder}/benchmark_all_results/${benchmark}_${end_commit}_${config}.json" --github 2>/dev/null)
+    echo ""
+    echo ""
+    echo -n "**${benchmark} - "
+    case "${config}" in
+      "st") echo -n "single-threaded" ;;
+      "mt") echo -n "multi-threaded (${num_mt_clients} clients)" ;;
+    esac
+    echo "**"
+    echo "<details>"
+    echo "<summary>"
+    echo "Sum of average per-item runtime: TODO,"
+    echo "$output" | grep 'geometric mean' | sed -E 's/^[+-]//' | sed 's/geometric mean//; s/[ |]//g; s/^/Geometric Mean: /'
```

Comment:
`  | grep 'Sum' | sed -E 's/^[+-]//' | awk -F"|" '{print $2, $6}' | sed 's/Sum//; s/[ ]//g; s/^/Sum of avg. query runtimes: /'
` | grep 'Geomean' | sed -E 's/^[+-]//' | sed 's/Geomean//; s/[ |]//g; s/^/Geometric Mean: /'`

???